src/coprocessor/dag/expr/builtin_encryption.rs

```diff
@@ -164,14 +221,86 @@ impl ScalarFunc {
         }
         Ok(Some(i64::from(LittleEndian::read_u32(&input[0..4]))))
     }
+
+    pub fn random_bytes<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &[Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let len = try_opt!(self.children[0].eval_int(ctx, row));
+        if len < 1 || len > 1024 {
+            Err(Error::overflow("length", "random_bytes"))
+        } else {
+            Ok(Some(Cow::Owned(random_bytes(len as usize))))
+        }
+    }
+
+    pub fn aes_encrypt<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &[Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let plaintext = try_opt!(self.children[0].eval_string(ctx, row));
+        let key = try_opt!(self.children[1].eval_string(ctx, row));
+        let iv = try_opt!(self.children[2].eval_string(ctx, row));
+        let mode = try_opt!(self.children[3].eval_int(ctx, row)) as i32;
```

Comment:
e, it seems different from tidb's implementation @lonng could you help to review it?

Reply:
@zz-jason Could you help to review it?